File: frontend/src/utils/typeMapping.js
export const TYPE_MAPPING = {
    'Normal': '노말',
    'Fire': '불꽃',
    'Water': '물',
    'Electric': '전기',
    'Grass': '풀',
    'Ice': '얼음',
    'Fighting': '격투',
    'Poison': '독',
    'Ground': '땅',
    'Flying': '비행',
    'Psychic': '에스퍼',
    'Bug': '벌레',
    'Rock': '바위',
    'Ghost': '고스트',
    'Dragon': '드래곤',
    'Dark': '악',
    'Steel': '강철',
    'Fairy': '페어리'
  };
  
  export const REVERSE_TYPE_MAPPING = Object.fromEntries(
    Object.entries(TYPE_MAPPING).map(([eng, kor]) => [kor, eng])
  );

File: backend/utils/type_mapping.py
TYPE_MAPPING = {
    '노말': 'Normal',
    '불꽃': 'Fire',
    '물': 'Water',
    '전기': 'Electric',
    '풀': 'Grass',
    '얼음': 'Ice',
    '격투': 'Fighting',
    '독': 'Poison',
    '땅': 'Ground',
    '비행': 'Flying',
    '에스퍼': 'Psychic',
    '벌레': 'Bug',
    '바위': 'Rock',
    '고스트': 'Ghost',
    '드래곤': 'Dragon',
    '악': 'Dark',
    '강철': 'Steel',
    '페어리': 'Fairy'
}

def get_korean_type(eng_type):
    reverse_mapping = {v: k for k, v in TYPE_MAPPING.items()}
    return reverse_mapping.get(eng_type, eng_type)

def get_english_type(kor_type):
    return TYPE_MAPPING.get(kor_type, kor_type)